prov/verbs/src/verbs_info.c

```diff
@@ -645,6 +769,17 @@ static int fi_ibv_alloc_info(struct ibv_context *ctx, struct fi_info **info,
 	fi->tx_attr->caps = ep_dom->caps;
 	fi->rx_attr->caps = ep_dom->caps;
 
+	if (!(fi->nic = ofi_nic_dup(NULL))) {
```

Comment:
Please avoid assignments inside if statements.  Some compilers still complain about this.